doc/man7/EVP_PKEY-ML-KEM.pod

```diff
@@ -16,37 +16,65 @@ EVP_PKEY-ML-KEM
 The B<ML-KEM-512>, B<ML-KEM-768>, and B<ML-KEM-1024> keytypes are implemented
 in OpenSSL's default and FIPS providers.
 
-=for comment (TODO(ML-KEM): Add FIPS support).
-
 =head2 Keygen Parameters
 
-By default, no parameters are required for generating a key pair.
+No mandatory parameters are required for generating a key pair.
 
 =over 4
 
 =item "seed" (B<OSSL_PKEY_PARAM_ML_KEM_SEED>) <octet string>
 
 Internally, ML-KEM generates keys using a 64-byte random value (seed), which is
 the concatenation of the 32-byte I<d> and I<z> parameters described in FIPS 203.
-The optional parameter can be used to set a pre-determined seed prior to
+This optional parameter can be used to set a pre-determined seed prior to
 keypair generation.
-According to FIPS 203, section 3.3, this parameter should only be used for test
-purposes and be treated with the same care as private key material.
 
-This parameter is only settable.
+Generated keys default to retaining the seed used.
+The seed is also by default retained when keys are loaded from B<PKCS#8> files
+in the seed format.
+When available, the seed parameter is also used during key export and import,
+with keys regenerated from the seed even if separately provided on import.
+
+When the seed is retained, it is also available as a B<gettable> parameter,
+and private key output to B<PKCS#8> files will be in seed format.
+When the seed is not available, because not included in the B<PKCS#8> file, not
+available on import, or not retained, B<PKCS#8> private key files will have the
+private key in FIPS 203 C<dk> format, without DER octet-string wrapping.
+
+Retention of the seed can be disabled by setting the C<pkey_seed_retain>
+parameter of the C<default> provider to C<no>.
+See the description of the B<-provparam> option in L<openssl(1)> to learn
+how to set provider options in the command line tools.
+See L<OSSL_PROVIDER_add_conf_parameter(3)> to learn how to set provider
+configuration options programmatically.
+When using the FIPS provider, the parameter may need to be set in the C<base>
+provider instead (or perhaps in all providers) if the intent is to retain seeds
+read from PKCS#8 files.
+
+In addition to its use in key generation, this parameter is also gettable and
+settable.
 
 See L<provider-keymgmt(7)/Common Information Parameters> for further
 information.
 
+=item "retain_seed" (B<OSSL_PKEY_PARAM_ML_KEM_RETAIN_SEED>) <int>
+
+This parameter is used only in key generation.
+When this parameter is set to a nonzero value, the seed is retained during key
+generation, when set to 0, the seed is not retained.
+When this parameter is simply omitted, and the C<pkey_retain_seed> provider
```

Comment:
By default == When this parameter is simply omitted,

Reply:
Another way of saying this is that setting the 'retain_seed' overrides the config option.

Reply:
Should this be a per algorithm config option rather than be called pkey_retain_seed

Reply:
In the case of `pkey(1)`, being a one-shot command that processes one key, it does not matter of course, but in the configuration file, yes, there a reasonable question about whether it should be possible to be more specific. It may be appropriate to have both a default, and per-algorithm settings?

Thoughts?

Reply:
I'll change this to a parameter that is *family*-specific. Any objections to or suggestions other than `ml-kem/retain_seed=(yes|no)`?